### PR TITLE
Added the ability to skip sheet loads

### LINKED ISSUE #1352: Add the ability to exclude/skip sheets in `load_study`
### Feature Inspiration

The load tests in the rabinowitz data repo (for 3 studies) are significantly slower than the rest.  `cold_exposure` is the current winner at 2h20m to complete.  (This is excluding the mzXML load.)  This results in the action status page showing 4.75 "billable" hours, but regardless, the tests should be fast.  It would all be MUCH faster if you could skip the load of the `Peak Annotation Files` sheet (i.e. the `PeakAnnotationFilesLoader`).

### Feature Description

This minimum feature suggestion is to add an `--exclude` option that takes a series of enumerated values.  We shouldn't use sheet names, because those can be customized and the value should be the same for all studies to work in the github action.

Note, this will mean that some problems might be overlooked, but that's already the case with `mzXML` files.  However, it could be implemented such that the skipped sheet is still checked for simple things like missing required values, but that would be a separate issue.  For this issue, just completely skip that loader.

### Alternatives

An alternative would be to be able to specify the actual sheets to load instead of which sheets to skip.  That could be a separate issue.  The options could co-exist and be mutually exclusive.

### Dependencies

None

### Comment

None

-----

### Assumptions

None

### Limitations

None

### Affected Components

None provided

### Requirements

None provided

#### Interface Change description

None provided

#### Code Change Description

None provided

#### Tests

None provided

### PR DESCRIPTION
**Pull request by hepcat72**
## Summary Change Description

This is intended to be used by the github test action in the rabinowitz data repo to dramatically cut down on test time by skipping the Peak Annotation Files sheet.

Details:

- Checked the `StudyLoader.get_loader_class` method name to `get_derived_class` to not conflate it with the contained loader classes.
- Added argument `--exclude-sheets` to `load_study`
- Added class method: `get_loader_classes` to `StudyLoader`.
- Added method `check_exclude_sheets` to `StudyLoader`.
- Added `exclude_sheets` to the instance and skipped loaders with a `DataSheetName` that matches a name in that list.

## Affected Issues/Pull Requests

- Resolves #1352
- Merges into main

## Review Notes

See comments in-line.

## Checklist

This pull request will be merged once the following requirements are met.  The
author and/or reviewers should uncheck any unmet requirements:

- Review requirements
  - Minimum approvals: 1 
  - No changes requested
  - All blocking issues resolved by reviewers
  - Specific reviewers: @__add_username_here__
    
  - Review period: 2 days 
- Associated issue/pull request requirements:
  
  - [x] All requirements in affected issues marked "resolved" are satisfied
  - [x] All required pull requests are merged *(or none)*
- Basic requirements
  
  - [x] [All linters pass](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#linting)
  - [x] [All tests pass](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#quality-control)
  - [x] All conflicts resolved
- Overhead requirements
  
  - [x] [New/changed method tests implemented/updated *(or no change)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#test-implementation)
  - [ ] [Updated *Unreleased* section of `changelog.md` *(or no change)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/changelog.md)
  - [x] [Migrations created & committed *(or no change)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#migration-process)

### CODE REVIEW

File: DataRepo/loaders/__init__.py [hunk 1]
```diff
@@ -46,6 +47,7 @@
     "SampleTableLoader",
     "SequencesLoader",
     "StudiesLoader",
+    "StudyLoader",
```
Comment (hepcat72) [inline DataRepo/loaders/__init__.py:50]: C. This should have been added before.

File: DataRepo/loaders/study_loader.py [hunk 2]
```diff
@@ -473,7 +512,13 @@ def load_data(self):
         for loader_key in self.Loaders._fields:
             if loader_key not in loaders.keys():
                 continue
+
             loader: TableLoader = loaders[loader_key]
+
+            # Skip loads that the user has excluded, based on a match of the class attribute DataSheetName
+            if loader.DataSheetName in self.exclude_sheets:
+                continue
```
Comment (hepcat72) [inline DataRepo/loaders/study_loader.py:520]: C. This is what actually skips the load.

File: DataRepo/loaders/study_loader.py [hunk 3]
```diff
@@ -1069,7 +1123,7 @@ def get_supported_versions(cls) -> List[str]:
         return [str(subcls.version_number) for subcls in StudyLoader.__subclasses__()]
 
     @classmethod
-    def get_loader_class(cls, df_dict, version=None):
+    def get_derived_class(cls, df_dict, version=None):
```
Comment (hepcat72) [inline DataRepo/loaders/study_loader.py:1126]: C. I realized that this name conflated contained loaders and subclasses, so I changed its name.

File: DataRepo/management/commands/load_study.py [hunk 4]
```diff
@@ -59,6 +59,18 @@ def add_arguments(self, parser):
             required=False,
         )
 
+        parser.add_argument(
+            "--exclude-sheets",
+            type=str,
+            help=(
+                f"[all] {{{[lc.DataSheetName for lc in StudyLoader.get_loader_classes()]}}} Load all sheets except "
+                "those supplied here.  (Use default sheet names in place of any custom names.)"
```
Comment (hepcat72) [inline DataRepo/management/commands/load_study.py:67]: C. Note, there's a TODO comment in code in an above file to add support for custom sheet names.

**Review by narzouni — APPROVED** [review]

**Pull request merged**